src/plugins/data/public/field_formats_provider/field_formats.ts

```diff
@@ -217,10 +223,26 @@ export class FieldFormatRegisty {
   }
 
   register = (fieldFormats: IFieldFormatType[]) => {
-    fieldFormats.forEach(fieldFormat => {
-      this.fieldFormats.set(fieldFormat.id, fieldFormat);
-    });
+    fieldFormats.forEach(fieldFormat => this.fieldFormats.set(fieldFormat.id, fieldFormat));
 
     return this;
   };
+
+  private decorateFieldFormatWithParams = (formatId: IFieldFormatId): Function | undefined => {
+    const concreteFieldFormat = this.fieldFormats.get(formatId);
+    const decorateMetaParams = (customOptions: Record<string, any> = {}) => ({
+      parsedUrl: {
+        origin: window.location.origin,
+        pathname: window.location.pathname,
+        basePath: this.basePath,
+      },
+      ...customOptions,
+    });
+
+    if (concreteFieldFormat) {
+      return function(params: Record<string, any> = {}, getConfig?: Function) {
+        return new concreteFieldFormat(decorateMetaParams(params), getConfig);
+      };
+    }
+  };
```

Comment:
I like the approach with the decorator, because it keeps everything else really clean. Would it be possible to add a comment explaining why this is necessary and how it helps us? Just so its purpose remains clear in the future.

Reply:
Signed off